<launch>
 	<arg name="veh"/>
 	<arg name="local" default="true" doc="true for running everything you possibly can on laptop. false for everything running on vehicle."/>
	<arg name="param_file_name" default="default" doc="Specify a param file. ex:megaman." />
 	<arg name="camera" default="true" doc="camera set to false means only apriltags no camera nodes are launched"/>
        <arg name="config" default="baseline" doc="Specify a config."/>
    <arg name="camera_name" default="atlas" doc="location of the camera" />
    <arg name="live" default="true" doc="live set to false means we don't run the actual camera (i.e. running from a log file" />
    <arg name="width" default="640" doc="width of compressed image"/>
    <arg name="height" default="480" doc="width of compressed image"/>  
    <arg name="tango" default="false" doc="subscribe tango image"/>      

    <!--   
 	<include if="$(arg camera)" file="$(find text_spotting_code)/launch/r200_camera.launch">
 	</include>
    --> 
    
     <!-- Camera -->
        <include if="$(arg camera)" file="$(find duckietown)/launch/camera.launch">
                <arg name="veh" value="$(arg veh)"/>
                <arg name="local" value="$(arg local)"/>
                <arg name="config" value="$(arg config)"/>
                <arg name="param_file_name" value="$(arg param_file_name)"/>
                <arg name="raw" value="true"/>
                <arg name="cam_info" value="true"/>
                <arg name="live" value="$(arg live)"/>
                <arg name="rect" value="true" />
        </include>

 	<!-- AprilTags Detections -->i
    <remap from="image_rect" to="camera_node/image/rect" />
    <remap from="camera_node/image/camera_info" to="camera_node/raw_camera_info" />
    <include file="$(find apriltags_ros_caffe)/launch/apriltag_detector_node.launch">
        <arg name="veh" value="$(arg veh)"/>
        <arg name="local" value="$(arg local)"/>
        <arg name="param_file_name" value="$(arg param_file_name)"/>
    </include>
 
 	<!-- Postprocessing node -->
 	<!--remap from="apriltags_postprocessing_node/apriltags_in" to="tag_detections" /-->
 	<!--include file="$(find apriltags_ros_caffe)/launch/apriltags_postprocessing_node.launch">
 		<arg name="veh" value="$(arg veh)"/>
        <arg name="local" value="$(arg local)"/>
        <arg name="param_file_name" value="$(arg param_file_name)"/-->
 	<!--/include-->
    <group ns="$(arg veh)">

        <!-- Run on local (laptop) -->
        <node if="$(arg local)" pkg="text_spotting_code" type="ncs_node.py" name="ncs_node" output="screen" clear_params="true" required="true">
                <param name="camera_name" value="$(arg camera_name)"/>
        </node>

        <!-- Run on remote (vehicle) -->
        <include unless="$(arg local)" file="$(find duckietown)/machines"/>
        <node unless="$(arg local)" machine="$(arg veh)" pkg="text_spotting_code" type="ncs_node.py" name="ncs_node" output="screen" clear_params="true" required="true">
                <param name="camera_name" value="$(arg camera_name)"/>
        </node>
    </group>


</launch>
